### PR TITLE
damlc inspect: Add output modes with less details

### PR DESCRIPTION
**Pull request by hurryabit**
This PR adds two new detail levels to `damlc inspect`:

* `--detail=-1` omits all package ids from the pretty printed output.
* `--detail=-2` omits all package ids, type information and kind
  annotations from the output.

We also slightly change the existing detail level `--detail=0`, which
is the default when the `--detail` flag is omitted, to make the output
more consistent:

* Kind annotations are no longer omitted when the kind is `*` (star).
* All location information is now omitted. To get location information,
  use `--detail=1`.

This is part of https://github.com/digital-asset/daml/issues/5756.

CHANGELOG_BEGIN
CHANGELOG_END

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

---
This change is [Reviewable](https://reviewable.io/reviews/digital-asset/daml/6280)

### CODE REVIEW

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs [hunk 1]
```diff
@@ -294,13 +308,16 @@ precBinding = 0
 precParam = 1
 
 pPrintAndKind :: Pretty a => PrettyLevel -> Rational -> (a, Kind) -> Doc ann
-pPrintAndKind lvl prec (v, k) = case k of
-    KStar -> pPrintPrec lvl 0 v
-    _ -> maybeParens (prec > precBinding) (pPrintPrec lvl 0 v <-> docHasType <-> kind_ (pPrintPrec lvl 0 k))
+pPrintAndKind lvl prec (v, k)
+  | levelHasKinds lvl = maybeParens (prec > precBinding) $
```
Comment (cocreature) [inline compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs:312]: Your PR description says that you no longer emit kind annotations if the kind is start but this change seems to be doing exactly the opposite? We special-cased star before and did not emit the kind annotation whereas now we no longer special-case it.

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs [hunk 2]
```diff
@@ -448,10 +467,14 @@ instance Pretty Expr where
       let (bs, e1) = view (rightSpine (unlocate _ETmLam)) e0
       in  hang (docTmLambda <> hsep (map (pPrintAndType lvl precParam) bs) <> docTmLambdaDot)
             2 (pPrintPrec lvl precELam e1)
-    e0@ETyLam{} -> maybeParens (prec > precELam) $
-      let (ts, e1) = view (rightSpine (unlocate _ETyLam)) e0
-      in  hang (docTyLambda <> hsep (map (pPrintAndKind lvl precParam) ts) <> docTyLambdaDot)
+    e0@ETyLam{}
```
Comment (cocreature) [inline compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs:470]: The description above mentions that you omit type annotations. Emitting type applications is another step imho. Not opposed to it but maybe update the comment.

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs (hunk 1 above)
Comment (hurryabit) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs:312]: Looks like there are too many negations in my PR description. It says
> * Kind annotations are no longer omitted when the kind is `*` (star).

Meaning, we now emit kind annotations even when the kind is `*`.

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs (hunk 2 above)
Comment (hurryabit) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs:470]: Good point. I should say we omit type information in level -2. It's not only type annotations but also type applications and type abstraction.

File: compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs (hunk 1 above)
Comment (cocreature) [reply compiler/daml-lf-ast/src/DA/Daml/LF/Ast/Pretty.hs:312]: I misread omitted as emitted :facepalm:

**Pull request merged**